packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -675,6 +719,20 @@ class Scaffold extends StatefulWidget {
     return scaffoldScope.geometryNotifier;
   }
 
+  /// Sets the [ScaffoldGeometry.floatingActionButtonNotchMaker] for the closest
+  /// [Scaffold] ancestor of the given context if one exist.
+  ///
+  /// Returns true if the [ScaffoldGeometry.floatingActionButtonNotchMaker] was
+  /// set.
+  /// Returns false if there as no [Scaffold] ancestor.
```

Comment:
this should talk about the contract, e.g. that you must call this with null if you are deactivated.

...having said that, I just thought of a problem with the setting to null.

Imagine you replace the scaffold's fab slot is a column with two children, and the fab is in the second child, and then you update the scaffold so that it's fab slot is a column with two children, but the fab is now in the first child.

The new fab will be built and will call `setFloatingActionButtonNotchMakerFor(this...)`, then the old fab will be deactivated and will call `setFloatingActionButtonNotchMakerFor(null)`.

Reply:
Replaced this with a closeable pattern. And extended the docs.